src/Sulu/Component/Content/Document/Subscriber/CopyLocaleSubscriber.php

```diff
@@ -0,0 +1,140 @@
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            Events::COPY_LOCALE => 'handleCopyLocale',
+        ];
+    }
+
+    public function handleCopyLocale(CopyLocaleEvent $event): void
+    {
+        $document = $event->getDocument();
+
+        if (!$document instanceof UuidBehavior) {
+            return;
+        }
+
+        $destLocale = $event->getDestLocale();
+        $uuid = $document->getUuid();
+
+        $webspaceKey = null;
+        $resourceLocatorStrategy = null;
+        if ($document instanceof WebspaceBehavior) {
+            $webspaceKey = $document->getWebspaceName();
+            $resourceLocatorStrategy = $this->resourceLocatorStrategyPool->getStrategyByWebspaceKey($webspaceKey);
+        }
+
+        $parentUuid = null;
+        if ($document instanceof ParentBehavior) {
+            $parentDocument = $this->documentInspector->getParent($document);
+
+            if (null !== $parentDocument) {
+                $parentUuid = $this->documentInspector->getUuid($parentDocument);
+            }
+        }
+
+        $destDocument = $this->documentManager->find(
+            $uuid,
+            $destLocale
+        );
+
+        if ($destDocument instanceof LocaleBehavior) {
+            $destDocument->setLocale($destLocale);
+        }
+
+        if ($destDocument instanceof TitleBehavior && $document instanceof TitleBehavior) {
+            $destDocument->setTitle($document->getTitle());
+        }
+
+        if ($destDocument instanceof StructureBehavior && $document instanceof StructureBehavior) {
+            $destDocument->setStructureType($document->getStructureType());
+            $destDocument->getStructure()->bind($document->getStructure()->toArray());
+        }
+
+        if ($destDocument instanceof WorkflowStageBehavior) {
+            $documentAccessor = new DocumentAccessor($destDocument);
+            $documentAccessor->set(WorkflowStageSubscriber::PUBLISHED_FIELD, null);
+        }
+
+        if ($destDocument instanceof ExtensionBehavior && $document instanceof ExtensionBehavior) {
+            $destDocument->setExtensionsData($document->getExtensionsData());
+        }
+
+        // TODO: This can be removed if RoutingAuto replaces the ResourceLocator code.
+        if ($destDocument instanceof ResourceSegmentBehavior
+            && $destDocument instanceof TitleBehavior
+            && null !== $resourceLocatorStrategy
+            && null !== $parentUuid
+            && null !== $webspaceKey) {
+            $resourceLocator = $resourceLocatorStrategy->generate(
+                $destDocument->getTitle(),
+                $parentUuid,
+                $webspaceKey,
+                $destLocale
+            );
+
+            $destDocument->setResourceSegment($resourceLocator);
+        }
+
+        $this->documentManager->persist($destDocument, $destLocale);
```

Comment:
@luca-rath havent we said that there should be an option to disable the domain event? will that be implemented in the follow up PR?

Reply:
exactly